src/exception/assembly_esp32.rs

```diff
@@ -483,13 +483,17 @@ unsafe extern "C" fn __default_naked_exception() {
         l32i    a6, sp, +XT_STK_EXCCAUSE  // put cause in a6 = a2 in callee
         beqi    a6, 4, .Level1Interrupt
 
+        movi    a0, (PS_INTLEVEL_EXCM | PS_WOE)
+        wsr     a0, PS
+        rsync
+
         mov     a7, sp                    // put address of save frame in a7=a3 in callee
         call4   __exception               // call handler <= actual call!
 
         j       .RestoreContext
 
         .Level1Interrupt:
-        movi    a0, (1 | PS_WOE)
+        movi    a0, (PS_INTLEVEL_EXCM | PS_WOE)
```

Comment:
I've been reading about `PS_INTLEVEL_EXCM`, or more specifically `PS.EXCM`. It seems from my understanding `PS.EXCM` is only ever set in two ways, manually by writing to `PS`, or in the case of a DoubleException (chapter 4, page 98 of the isa rm).  When `PS.EXCM` is set, it overwrites some fields, namely `INTLEVEL` (see 4.4.2.4 of the isa rm), to some predefined threshold, hence the existence of this variable.

Seeing as we don't ever set `PS.EXCM` in our code, and in the case of a double fault we will abort, should we be using `PS_INTLEVEL_EXCM`? I guess this is answering the TODO next the definition in `global_asm!`, I _think_ we can actually use `XCHAL_NUM_INTLEVELS` (which by the way is up to 6 for all esp32 chips, and only 2 for esp8266) instead here. What do you think?

Reply:
that what I was thinking when writing the TODO

but I'm not sure anymore if that is okay - looking at _4.4.4.3 The Level-1 Interrupt Process_ and _4.4.5.3 The High-Priority Interrupt Process_ - especially 

> Thus, all interrupts at level EXCMLEVEL and below are masked during the time PS.EXCM=1. This is done to allow high-level language coding with the Windowed Register Option of interrupt handlers for interrupts whose level is not greater than EXCMLEVEL.

Reply:
Ah yes, I actually remember reading something along the lines of that before (can't quite remember where). If I recall correctly, I think higher level interrupt levels shouldn't expect to even have a stack available so its not very useful for us in the Rust world.

Let's change the comment and be done with it :)